Bottom: a1d9ab0d1d177ca7fdf5336aa9b00ea760c62ba3
Top:    50f815026778a39dac6ff1b288fa6aa2230f23ff
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-21 12:21:11 +0800

Refresh of autofs-5.0.5-wait-for-master-map-available-at-start.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index 4f5d0ca..cdbc226 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1924,11 +1924,13 @@ static int do_master_read_master(struct master *master, time_t age, int wait)
 		if (nanosleep(&t, NULL) == -1)
 			break;
 
-		elapsed += retry_wait;
-		if (max_wait > 0 && elapsed >= max_wait) {
-			logmsg("problem reading master map, "
-				"maximum wait exceeded");
-			break;
+		if (max_wait > 0) {
+			elapsed += retry_wait;
+			if (elapsed >= max_wait) {
+				logmsg("problem reading master map, "
+					"maximum wait exceeded");
+				break;
+			}
 		}
 	}
